src/Shared/TransportationDeviceClient.cs

```diff
@@ -79,9 +76,36 @@ public async Task<dynamic> GetDynamicDigitalTwinAsync()
             return dyn;
         }
 
-        public Task RegisterDesiredPropertyUpdateCallbackAsync(MethodCallback methodHandler)
+        // this method will process any desired property change events
+        private static async Task OnDesiredPropertyChanged(TwinCollection desiredProperties, object userContext)
+        {
+            List<Task> taskList = new List<Task>();
+
+            // add each handler to a task list for processing
+            foreach (DesiredPropertyUpdateCallback callbackMethod in desiredCallbacks)
+            {
+                taskList.Add(callbackMethod(desiredProperties, userContext));
+            }
+
+            // execute all tasks and wait for them to complete
+            Task.WaitAll(taskList.ToArray()); 
+
+            // return to caller
+            return;
+        }
+
+        // this method lets you add a new desired property callback method to the internal array
+        public async void RegisterDesiredPropertyUpdateCallbackAsync(DesiredPropertyUpdateCallback callbackHandler)
```

Comment:
How would the caller monitor and control the execution of the method ?

Reply:
Not sure I understand this? the 'host' for the callback method doesn't control the execution. its done by IOT Hub sending an event back down to the device. This triggers the callback to execute.